### PR TITLE
InstallNode Script: use correct version (oss vs ent)

### LINKED ISSUE #19542: Install Wizards and Discover should match Teleport installation to OSS or Enterprise types
Expected behavior:

The install on the target machine for the service should be the OSS or enterprise type matching to the server.

Current behavior:

The OSS version of Teleport is installed whether or not the Proxy/Auth are enterprise.

Bug details:
- Teleport version: 11.1.3
- Recreation steps

Use the Add server on a enterprise or cloud version.

### PR DESCRIPTION
**Pull request by marcoandredinis**
By default install the OSS version.
However, if the cluster is enterprise, then install the `teleport-ent` package instead.

Demo
Script output when using an Enterprise build:
```
root@tmp01:/# rg teleport-ent install-node.sh
778:        URL="https://get.gravitational.com/teleport-ent-v${TELEPORT_VERSION}-${TELEPORT_BINARY_TYPE}-${TELEPORT_ARCH}-centos6-bin.tar.gz"
780:        URL="https://get.gravitational.com/teleport-ent-v${TELEPORT_VERSION}-${TELEPORT_BINARY_TYPE}-${TELEPORT_ARCH}-bin.tar.gz"
805:    URL="https://get.gravitational.com/teleport-ent_${TELEPORT_VERSION}_${DEB_ARCH}.deb"
827:    URL="https://get.gravitational.com/teleport-ent-${TELEPORT_VERSION}-1.${RPM_ARCH}.rpm"
```

Fixes #19542

### CODE REVIEW

**Review by mdwn — APPROVED** [review]

File: lib/web/scripts/node-join/install.sh [hunk 1]
```diff
@@ -777,9 +777,9 @@ fi
 if [[ ${TELEPORT_FORMAT} == "tarball" ]]; then
     # handle centos6 URL override
     if [[ ${DISTRO_TYPE} == "centos6" ]]; then
-        URL="https://get.gravitational.com/teleport-v${TELEPORT_VERSION}-${TELEPORT_BINARY_TYPE}-${TELEPORT_ARCH}-centos6-bin.tar.gz"
+        URL="https://get.gravitational.com/{{.packageName}}-v${TELEPORT_VERSION}-${TELEPORT_BINARY_TYPE}-${TELEPORT_ARCH}-centos6-bin.tar.gz"
```
Comment (jakule) [inline lib/web/scripts/node-join/install.sh:780]: CentOS 6 is no longer supported by Teleport. I also think that the link now has `centos7` in the name. Should we update it?

Comment (marcoandredinis) [reply lib/web/scripts/node-join/install.sh:780]: I think that should be a separate issue, don't you think?
Seems like we already support other `centos*` distro versions.
However you said that the link has `centos7` in the name, which is not present in the script.

Well, happy to tackle this one, but I think we should have a separate issue for it

**Review by ryanclark — APPROVED** [review]

File: lib/web/scripts/node-join/install.sh (hunk 1 above)
Comment (jakule) [reply lib/web/scripts/node-join/install.sh:780]: I missed the `DISTRO_TYPE` check. So, here is some background story. We used to support centos 6, and we had a special build for it. I think that with Teleport 9, we dropped support for CentOS 6, and then we changed the name of our tar to `centos7`. Later we decided that having two packages (one for CentOS 7 and one for everyone else) is too much work, and currently, we're releasing only one package that works everywhere.
Basically, this is dead code as CentOS 6 is not supported anymore, and we don't release packages with `centos6` in the name. I created an issue to not forget about it https://github.com/gravitational/teleport/issues/20813

**Review by jakule — APPROVED** [review]

File: lib/web/scripts/node-join/install.sh (hunk 1 above)
Comment (marcoandredinis) [reply lib/web/scripts/node-join/install.sh:780]: Thank you!
